src/components/permissionable/permissionable.cairo

```diff
@@ -1,3 +1,102 @@
 // permissionable component
+// *************************************************************************
+//                              PERMISSIONABLE COMPONENT
+// *************************************************************************
+#[starknet::component]
+pub mod PermissionableComponent {
+    // *************************************************************************
+    //                              IMPORTS
+    // *************************************************************************
 
+    use starknet::storage::{Map, StorageMapReadAccess, StorageMapWriteAccess};
+    use starknet::{ContractAddress, get_caller_address, get_block_timestamp};
+    use token_bound_accounts::components::account::account::AccountComponent;
+    use token_bound_accounts::interfaces::IAccount::{IAccount, IAccountDispatcherTrait};
+    use token_bound_accounts::components::account::account::AccountComponent::InternalImpl;
+    use token_bound_accounts::interfaces::IPermissionable::{
+        IPermissionable, IPermissionableDispatcher, IPermissionableDispatcherTrait
+    };
+
+
+    #[storage]
+    pub struct Storage {
+        permissions: Map<ContractAddress, (ContractAddress, bool)> // <owner => <caller, bool>>
```

Comment:
just realizing this mapping is wrong.

You are trying to check if a particular caller is granted permissions by an owner. it should be:

```
permissions: Map<(ContractAddress, ContractAddress), bool>
```

can lead to serious vulnerability as seen in the `has_permission` function